client/src/pages/Profile/ProfileAvailability/ProfileAvailability.tsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect, useState } from 'react';
+import { format, getDate, getDay, getMonth } from 'date-fns';
+import { DateRange, Range } from 'react-date-range';
+import 'react-date-range/dist/styles.css'; // main style file
+import 'react-date-range/dist/theme/default.css'; // theme css file
+import { Box, Button, Card, CardContent, Divider, Grid, Icon, Switch, TextField, Typography } from '@material-ui/core';
+import DateRangeIcon from '@material-ui/icons/CalendarToday';
+import SaveIcon from '@material-ui/icons/Save';
+import { eachDayOfInterval } from 'date-fns/esm';
+import { getUserProfile, updateProfile } from '../../../helpers/APICalls/profile';
+import { AvailabilityInDays } from '../../../interface/Profile';
+import MONTHS from '../../../lib/constants/MONTHS.json';
+import DAYS from '../../../lib/constants/DAYS.json';
+
+import useStyles from './useStyles';
+
+function ProfileAvailability(): JSX.Element {
+  const classes = useStyles();
+
+  const today = new Date();
+  const nextDay = new Date(today);
+  nextDay.setDate(nextDay.getDate() + 1);
+
+  const [hidden, setHidden] = useState(true);
+  const [selection, setSelection] = useState<Range>({
+    startDate: today,
+    endDate: nextDay,
+    key: 'rollup',
+  });
+
+  const [availability, setAvailability] = useState<AvailabilityInDays>({
+    monday: false,
+    tuesday: false,
+    wednesday: false,
+    thursday: false,
+    friday: false,
+    saturday: false,
+    sunday: false,
+  });
+
+  useEffect(() => {
+    async function getSitterAvailability() {
+      const userProfile = await getUserProfile();
+
+      if (userProfile.profile.availability !== null) {
+        setAvailability(userProfile.profile.availability);
+      }
+    }
+    getSitterAvailability();
+  }, []);
+
+  useEffect(() => {
+    async function updateAvailability() {
+      await updateProfile(availability);
+    }
+    updateAvailability();
```

Comment:
Prefer using an IFFE for this kind of implementation.
`
(async () => {
  await updateProfile(availability);
})();`